### PR TITLE
doc: pg_num should always be a power of two

### PR DESCRIPTION
**Pull request by capri1989**
Hi,

I updated the pg_num section in the docs just a little to be more strict.
I think we should make it crystal clear that pg num should be a power of two and only if there's a real good reason and someone knows exactly why they need a different value they can proceed. 

I think this is the first step of several to make this clearer to the end user. Next step would be to show a Health WARN and also to flag the pool creation red in the dashboard if someone would like to add a pool and the pg count isn't a power of two.

Why do I would like to be more strict about it? We've seen more and more clusters were the osd variation was really high. The fix to that was always to set the pg_num + balancer. Also the pg autoscaler only uses power of two so we shouldn't even encourage people to use something different right from the beginning or to phrase i differently, we should protect the end-user right away.

FIxes: https://tracker.ceph.com/issues/41004
Signed-off-by: Kai Wagner <kwagner@suse.com>

### CODE REVIEW

File: doc/rados/operations/placement-groups.rst [hunk 1]
```diff
@@ -433,9 +433,11 @@ You should then check if the result makes sense with the way you
 designed your Ceph cluster to maximize `data durability`_,
 `object distribution`_ and minimize `resource usage`_.
 
-The result should be **rounded up to the nearest power of two.**
-Rounding up is optional, but recommended for CRUSH to more evenly balance
-the number of objects among placement groups.
+The result should always be **rounded up to the nearest power of two.**
+The way the CRUSH algorithm is designed, a power of two leads to a more
```
Comment (liewegas) [inline doc/rados/operations/placement-groups.rst]: nit: it's actually rados, not crush, that figures out how PGs shard.  Maybe just ``A power of two ...``

File: doc/rados/operations/placement-groups.rst [hunk 2]
```diff
@@ -433,9 +433,11 @@ You should then check if the result makes sense with the way you
 designed your Ceph cluster to maximize `data durability`_,
 `object distribution`_ and minimize `resource usage`_.
 
-The result should be **rounded up to the nearest power of two.**
-Rounding up is optional, but recommended for CRUSH to more evenly balance
-the number of objects among placement groups.
+The result should always be **rounded up to the nearest power of two.**
+The way the CRUSH algorithm is designed, a power of two leads to a more 
+evenly balance number of objects among placement groups. You could set it
+to a different value but you should be aware that this could result in uneven
```
Comment (liewegas) [inline doc/rados/operations/placement-groups.rst]: ``...result in a more uneven distribution of data across your OSDs.``

**Comment by capri1989** [comment]
@liewegas Thanks for your feedback. Addressed both comments

File: doc/rados/operations/placement-groups.rst [hunk 3]
```diff
@@ -433,9 +433,10 @@ You should then check if the result makes sense with the way you
 designed your Ceph cluster to maximize `data durability`_,
 `object distribution`_ and minimize `resource usage`_.
 
-The result should be **rounded up to the nearest power of two.**
-Rounding up is optional, but recommended for CRUSH to more evenly balance
-the number of objects among placement groups.
+The result should always be **rounded up to the nearest power of two.**
+A power of two leads to a more evenly balance number of objects among placement groups.
```
Comment (liewegas) [inline doc/rados/operations/placement-groups.rst]: oops, s/balance/balanced/

Comment (capri1989) [reply doc/rados/operations/placement-groups.rst]: It seems I should better go to bed now ;-)

File: doc/rados/operations/placement-groups.rst [hunk 4]
```diff
@@ -433,9 +433,10 @@ You should then check if the result makes sense with the way you
 designed your Ceph cluster to maximize `data durability`_,
 `object distribution`_ and minimize `resource usage`_.
 
-The result should be **rounded up to the nearest power of two.**
-Rounding up is optional, but recommended for CRUSH to more evenly balance
-the number of objects among placement groups.
+The result should always be **rounded up to the nearest power of two.**
+A power of two leads to a more evenly balanced number of objects among placement groups. 
+You could set it to a different value but you should be aware that this could result in 
+a more uneven distribution of data across your OSDs.
```
Comment (l-mb) [inline doc/rados/operations/placement-groups.rst]: "a more uneven" -> "an uneven".

File: doc/rados/operations/placement-groups.rst [hunk 5]
```diff
@@ -433,9 +433,10 @@ You should then check if the result makes sense with the way you
 designed your Ceph cluster to maximize `data durability`_,
 `object distribution`_ and minimize `resource usage`_.
 
-The result should be **rounded up to the nearest power of two.**
-Rounding up is optional, but recommended for CRUSH to more evenly balance
-the number of objects among placement groups.
+The result should always be **rounded up to the nearest power of two.**
+A power of two leads to a more evenly balanced number of objects among placement groups.
```
Comment (l-mb) [inline doc/rados/operations/placement-groups.rst]: A power of two will evenly balance ...

File: doc/rados/operations/placement-groups.rst (hunk 4 above)
Comment (l-mb) [reply doc/rados/operations/placement-groups.rst]: "Use of such values should be limited to incrementally stepping from one power of two to another."

**Comment by jan--f** [comment]
We should probably backport this as well? If so please open a tracker ticket and mention it in your commit @capri1989.

**Comment by capri1989** [comment]
@l-mb comments addressed

File: doc/rados/operations/placement-groups.rst [hunk 6]
```diff
@@ -433,9 +433,11 @@ You should then check if the result makes sense with the way you
 designed your Ceph cluster to maximize `data durability`_,
 `object distribution`_ and minimize `resource usage`_.
 
-The result should be **rounded up to the nearest power of two.**
-Rounding up is optional, but recommended for CRUSH to more evenly balance
-the number of objects among placement groups.
+The result should always be **rounded up to the nearest power of two.**
+A power of two will evenly balance the number of objects among placement groups. 
+You could set it to a different value but you should be aware that this could result in 
+an uneven distribution of data across your OSDs. Use of such values shoud be limited to
```
Comment (krig) [inline doc/rados/operations/placement-groups.rst]: typo: `shoud` -> `should`

Comment (l-mb) [reply doc/rados/operations/placement-groups.rst]: Fixed

**Comment by theanalyst** [comment]
jenkins build docs

**Comment by theanalyst** [comment]
jenkins render docs

**Comment by theanalyst** [comment]
jenkins test docs

**Pull request merged**